cmd/spire-server/cli/federation/create.go

```diff
@@ -0,0 +1,138 @@
+package federation
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/mitchellh/cli"
+	trustdomainv1 "github.com/spiffe/spire-api-sdk/proto/spire/api/server/trustdomain/v1"
+	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"github.com/spiffe/spire/cmd/spire-server/util"
+	common_cli "github.com/spiffe/spire/pkg/common/cli"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	profileHTTPSWeb    = "https_web"
+	profileHTTPSSPIFFE = "https_spiffe"
+)
+
+// NewCreateCommand creates a new "create" subcommand for "federation" command.
+func NewCreateCommand() cli.Command {
+	return newCreateCommand(common_cli.DefaultEnv)
+}
+
+func newCreateCommand(env *common_cli.Env) cli.Command {
+	return util.AdaptCommand(env, new(createCommand))
+}
+
+// FederationRelationships type is used for parsing federation relationships from file
+type federationRelationships struct {
+	FederationRelationships []*federationRelationshipJSON `json:"federation_relationships"`
+}
+
+type createCommand struct {
+	path                  string
+	trustDomain           string
+	bundleEndpointURL     string
+	bundleEndpointProfile string
+	endpointSPIFFEID      string
+	bundleFormat          string
+	bundlePath            string
+}
+
+func (*createCommand) Name() string {
+	return "federation create"
+}
+
+func (*createCommand) Synopsis() string {
+	return "Creates a federation relationship to a foreign trust domain"
+}
+
+func (c *createCommand) AppendFlags(f *flag.FlagSet) {
+	f.StringVar(&c.path, "data", "", "Path to a file containing federation relationships in JSON format (optional). If set to '-', read the JSON from stdin.")
+	f.StringVar(&c.trustDomain, "trustDomain", "", `Name of the trust domain to federate with (e.g., example.org)`)
+	f.StringVar(&c.bundleEndpointURL, "bundleEndpointURL", "", "URL of the SPIFFE bundle endpoint that provides the trust bundle (must use the HTTPS protocol)")
+	f.StringVar(&c.bundleEndpointProfile, "bundleEndpointProfile", profileHTTPSSPIFFE, fmt.Sprintf("Endpoint profile type (either %q or %q)", profileHTTPSWeb, profileHTTPSSPIFFE))
+	f.StringVar(&c.endpointSPIFFEID, "endpointSpiffeID", "", "SPIFFE ID of the SPIFFE bundle endpoint server. Only used for 'spiffe' profile.")
+	f.StringVar(&c.bundlePath, "bundlePath", "", "Path to the bundle data (optional). Only used for 'spiffe' profile.")
```

Comment:
This comment is no longer accurate, also flag name change.
```suggestion
	f.StringVar(&c.bundlePath, "trustDomainBundlePath", "", "Path to the trust domain bundle data (optional).")
```